### ISSUE TITLE
Unnecessary logging when waiting for assignments is canceled.

### ISSUE THREAD

**Issue by Laremere** [googleforgames/open-match#1228]
Currently, when a waiting for assignment, a client will start the call and receive a message when the assignment has been updated.  Then it drops the call, because it doesn't need it anymore.

On the next iteration of the wait for assignment loop, it then logs in two places (at least) that the context is canceled:
https://github.com/googleforgames/open-match/blob/master/internal/statestore/redis.go#L584

https://github.com/googleforgames/open-match/blob/master/internal/statestore/redis.go#L228

This is partially a result of over logging that needs to be cleaned up as part of https://github.com/googleforgames/open-match/issues/1102

However, there's another issue here:
We've now stated that getting an assignment is a terminal state for a ticket (besides being deleted), so there's no point in the wait for assignment call to continue querying the database.  It will always get the same answer until it errors due to the ticket not being there, or canceled by the caller.

Should the FE change so that it will terminate the call once it successfully gets an assignment?  I think that behavioral change falls into the backwards compatibility promises for 1.0, but just barely.

Otherwise, perhaps we need to both: Fix the over logging that's occurring, and suppress logging for canceled context on wait for assignment, though that starts to get iffy on suppressing errors the user might care about.  We could get smart about when to suppress (eg, only after assignment was found), but that's getting even more magical and weird.

**Comment by akremsa** [comment]
Hello @Laremere 
I will take a look at this one.
